desc/optimize/optimizer.py

```diff
@@ -146,15 +147,45 @@ def optimize(  # noqa: C901 - FIXME: simplify this
             linear_constraints = maybe_add_self_consistency(linear_constraints)
         if len(linear_constraints):
             objective = LinearConstraintProjection(objective, linear_constraints)
+            if nonlinear_constraint is not None:
+                nonlinear_constraint = LinearConstraintProjection(
```

Comment:
this is meant to trigger the error in the init, bc this means that a nonlinear constraint was not caught and wrapped by the earlier maybe_wrap?

Reply:
no this is just doing the linear constraint projection on top of the nonlinear constraint, so that the objective and constraint both take the same reduced state vector.